backstory/src/main/java/com/google/sps/perspective/data/ContentDecisions.java

```diff
@@ -69,4 +71,104 @@ private static boolean isToxic(Map<AttributeType, Float> attributeTypesToScores)
+   *     API
+   * @return true, if sexually explicit score >= 60%; false, if not
+   * @throws IllegalArgumentException, if attributeTypesToScores is null or does not contain a
+   *     sexually explicit score
+   */
+  private static boolean isSexuallyExplicit(Map<AttributeType, Float> attributeTypesToScores)
+      throws IllegalArgumentException {
+    if (attributeTypesToScores == null) {
+      throw new IllegalArgumentException("Map (attributeTypesToScores) cannot be null.");
+    } else if (!attributeTypesToScores.containsKey(AttributeType.SEXUALLY_EXPLICIT)) {
+      throw new IllegalArgumentException(
+          "Map (attributeTypesToScores) does not contain a sexually explicit score");
+    }
+
+    float sexuallyExplicit = attributeTypesToScores.get(AttributeType.SEXUALLY_EXPLICIT);
+
+    return sexuallyExplicit >= .6f;
+  }
+
+  /**
+   * Private helper method to check if content is considered profane.
+   * Threshold for profane content is a score greater than or equal to 80%
+   * after experimenting with Perspective API
+   *
+   * @param attributeTypesToScores a map with attribute types mapped to scores from the Perspective
+   *     API
+   * @return true, if profanity score >= 80%; false, if not
+   * @throws IllegalArgumentException, if attributeTypesToScores is null or does not contain a
+   *     profanity score
+   */
+  private static boolean isProfane(Map<AttributeType, Float> attributeTypesToScores)
+      throws IllegalArgumentException {
+    if (attributeTypesToScores == null) {
+      throw new IllegalArgumentException("Map (attributeTypesToScores) cannot be null.");
+    } else if (!attributeTypesToScores.containsKey(AttributeType.PROFANITY)) {
+      throw new IllegalArgumentException(
+          "Map (attributeTypesToScores) does not contain a profanity score");
+    }
+
+    float profanity = attributeTypesToScores.get(AttributeType.PROFANITY);
+
+    return profanity >= .8f;
+  }
+
+  /**
+   * Private helper method to check if content is considered offensive.
+   * Threshold for offensive content is an identity attack score greater than or equal to 80%
+   * after experimenting with Perspective API
+   *
+   * @param attributeTypesToScores a map with attribute types mapped to scores from the Perspective
+   *     API
+   * @return true, if identity attack score >= 80%; false, if not
+   * @throws IllegalArgumentException, if attributeTypesToScores is null or does not contain a
+   *     identity attack score
+   */
+  private static boolean isOffensive(Map<AttributeType, Float> attributeTypesToScores)
+      throws IllegalArgumentException {
+    if (attributeTypesToScores == null) {
+      throw new IllegalArgumentException("Map (attributeTypesToScores) cannot be null.");
+    } else if (!attributeTypesToScores.containsKey(AttributeType.IDENTITY_ATTACK)) {
+      throw new IllegalArgumentException(
+          "Map (attributeTypesToScores) does not contain an identity attack score");
+    }
```

Comment:
There's some duplication here... this may be easier to maintain if moved to a helper method:

```
void validateHasScoreForAttributeType(Map<AttributeType, Float> attributeTypesToScores, AttributeType attributeType) throws IllegalArgumentException;
```

Reply:
i can do that. Thank you for the suggestion!

Reply:
Done